File: src/main/java/com/huangyueran/spark/sql/LoadSave.java
package com.huangyueran.spark.sql;

import com.huangyueran.spark.utils.Constant;
import com.huangyueran.spark.utils.SparkUtils;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.*;

/**
 * @author huangyueran
 * @category LoadSave
 * @time 2019-7-24 13:58:59
 */
public class LoadSave {
    public static void main(String[] args) {
        JavaSparkContext sc = SparkUtils.getRemoteSparkContext(LoadSave.class);

        // 创建DataFrame 读取json
        SQLContext sqlContext = new SQLContext(sc);

        DataFrameReader dataFrameReader = sqlContext.read();

        // parquet 是本地数据存储的格式
        Dataset<Row> dataset = dataFrameReader.load("/data/resources/users.parquet");

        dataset.printSchema();
        dataset.show();

        // 通过关writer写入并保存save
        DataFrameWriter write = dataset.select("name", "favorite_color").write();
        write.save("tmp/namesAndColors.parquet");

        Dataset<Row> load = dataFrameReader.load("tmp/namesAndColors.parquet/part-00000-c8ac19a7-9e13-49e2-a4d0-7cfe469bdfcf-c000.snappy.parquet");
        load.show();

        sc.close();
    }
}

File: src/main/scala/com/hyr/spark/sql/LoadSave.scala
package com.hyr.spark.sql

import com.huangyueran.spark.utils.Constant
import com.hyr.spark.utils.SparkUtils
import org.apache.spark.sql.SparkSession

/** *****************************************************************************
  * @date 2019-08-29 15:19
  * @author: <a href=mailto:huangyr>黄跃然</a>
  * @Description:
  * *****************************************************************************/
object LoadSave {

  def main(args: Array[String]): Unit = {
    val sparkContext = SparkUtils.getLocalSparkContext(LoadSave.getClass)

    val sparkSession = SparkSession.builder().appName("LoadSave").getOrCreate()

    val dataSet = sparkSession.read.load(Constant.LOCAL_FILE_PREX+"/data/resources/users.parquet")
    dataSet.printSchema
    dataSet.show

    val write = dataSet.select("name", "favorite_color", "favorite_numbers").write
    // 通过关writer写入并保存save
    write.save(Constant.LOCAL_FILE_PREX+"tmp/namesAndColors.parquet")



  }

}
